File: C++/Step09-Math_2 Prime and Geometric/1002.cpp
//
//  1002.cpp
//  Step9-Math_2 Prime and Geometric
//
//  Created by 김예빈 on 2019. 9. 5..
//  Copyright © 2019년 김예빈. All rights reserved.
//

#include <iostream>
using namespace std;

int main(int argc, const char * argv[]) {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int repeat, dist, diff, sum, x_1, y_1, r_1, x_2, y_2, r_2;
    
    cin >> repeat;
    
    while(repeat--) {
        cin >> x_1 >> y_1 >> r_1 >> x_2 >> y_2 >> r_2;
        
        dist = (x_1 - x_2) * (x_1 - x_2) + (y_1 - y_2) * (y_1 - y_2);
        diff = (r_1 - r_2) * (r_1 - r_2);
        sum = (r_1 + r_2) * (r_1 + r_2);
        
        if(dist == 0)
            if(diff == 0)
                cout << "-1\n";
            else
                cout << "0\n";
        
        else if(dist == sum || dist == diff)
            cout << "1\n";
        
        else if(diff < dist && dist < sum)
            cout << "2\n";
        
        else
            cout << "0\n";
    }

    return 0;
}

File: BaekJoon Online Judge/Swift/etc/1002.swift
//
//  1002.swift
//  Step09-Math_2 Prime and Geometric
//
//  Created by Yebin Kim on 2019/11/24.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

if let tc = Int(readLine()!) {
    for _ in 0..<tc {
        let input = readLine()!.split(separator: " ").map{ Int($0)! }
        let x1 = input[0]
        let y1 = input[1]
        let r1 = input[2]
        let x2 = input[3]
        let y2 = input[4]
        let r2 = input[5]
        
        let dist = (x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2)
        let diff = (r1 - r2) * (r1 - r2)
        let sum = (r1 + r2) * (r1 + r2)
        
        if dist == 0 {
            if diff == 0 {
                print(-1)
            } else {
                print(0)
            }
        } else if dist == sum || dist == diff {
            print(1)
        } else if(diff < dist && dist < sum) {
            print(2)
        } else {
            print(0)
        }
    }
}
